pyproject.toml

```diff
@@ -0,0 +1,99 @@
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [
+  "setuptools>=61.2"
+]
+
+[project]
+authors = [{name = "Matias Aguirre", email = "matiasaguirre@gmail.com"}]
+classifiers = [
+  "Development Status :: 4 - Beta",
+  "Environment :: Web Environment",
+  "Intended Audience :: Developers",
+  "License :: OSI Approved :: BSD License",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.7",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Topic :: Internet"
+]
+dependencies = [
+  "cryptography>=1.4",
+  "defusedxml>=0.5.0rc1",
+  "oauthlib>=1.0.3",
+  "PyJWT>=2",
+  "python3-openid>=3.0.10",
+  "requests>=2.9.1",
+  "requests-oauthlib>=0.6.1"
+]
+description = "Python social authentication made simple."
+dynamic = ["version", "readme"]
+keywords = [
+  "auth",
+  "oauth",
+  "openid",
+  "saml",
+  "social"
+]
+license = {text = "BSD"}
+name = "social-auth-core"
+requires-python = ">=3.7"
+
+[project.optional-dependencies]
+all = [
+  "cryptography>=2.1.1",
+  "python-jose>=3",
+  "python3-saml>=1.5"
+]
+allpy3 = [
+  "cryptography>=2.1.1",
+  "python-jose>=3",
+  "python3-saml>=1.5"
+]
+azuread = [
+  "cryptography>=2.1.1"
+]
+openidconnect = [
+  "python-jose>=3"
+]
+saml = [
+  "python3-saml>=1.5"
+]
+testing = [
+  "coverage>=3.6",
+  "cryptography>=2.1.1",
+  "httpretty>=0.9.6",
+  "pytest>=4.5",
+  "pytest-cov>=2.7.1",
+  "python-jose>=3",
+  "python3-saml>=1.5"
+]
```

Comment:
This all duplicates requirements-*.txt content. Can the dependencies be specified just once?

Reply:
Yes, but can we push that to a separate PR?  This one already touches a lot of files and `tox.ini`, and GitHub Action caching will need to be modified when dropping requirements files.

Reply:
You can generate the requirements files from the `pyproject.toml` one:

```
python3 -m piptools compile --output-file=requirements.txt pyproject.toml
python3 -m piptools compile --extra=dev --output-file=requirements-dev.txt pyproject.toml
etc
```